README.md

```diff
@@ -1,80 +1,740 @@
+contract Leverage {
+    Lending i_lending;
+    CornDEX i_cornDEX;
+    Corn i_corn;
+    address public owner;
+
+    event LeveragedPositionOpened(address user, uint256 loops);
+    event LeveragedPositionClosed(address user, uint256 loops);
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only the owner can call this function");
+        _;
+    }
+
+    constructor(address _lending, address _cornDEX, address _corn) {
+        i_lending = Lending(_lending);
+        i_cornDEX = CornDEX(_cornDEX);
+        i_corn = Corn(_corn);
+        // Approve the DEX to spend the user's CORN
+        i_corn.approve(address(i_cornDEX), type(uint256).max);
+    }
+    
+    /**
+     * @notice Claim ownership of the contract so that no one else can change your position or withdraw your funds
+     */
+    function claimOwnership() public {
+        owner = msg.sender;
+    }
+
+    /**
+     * @notice Open a leveraged position, recursively borrowing CORN, swapping it for ETH, and adding it as collateral
+     * @param reserve The amount of ETH that we will keep in the contract as a reserve to prevent liquidation
+     */
+    function openLeveragedPosition(uint256 reserve) public payable onlyOwner {
+        uint256 loops = 0;
+        while (true) {
+            // Write more code here
+            loops++;
+        }
+        emit LeveragedPositionOpened(msg.sender, loops);
+    }
+
+    /**
+     * @notice Close a leveraged position, recursively withdrawing collateral, swapping it for CORN, and repaying the lending contract until the position is closed
+     */
+    function closeLeveragedPosition() public onlyOwner {
+        uint256 loops = 0;
+        while (true) {
+            // Write more code here
+            loops++;
+        }
+        emit LeveragedPositionClosed(msg.sender, loops);
+    }
+
+    /**
+     * @notice Withdraw the ETH from the contract
+     */
+    function withdraw() public onlyOwner {
+        uint256 balance = address(this).balance;
+        require(balance > 0, "No balance to withdraw");
+        (bool success, ) = payable(msg.sender).call{value: balance}("");
+        require(success, "Failed to send Ether");
+    }
+
+    receive() external payable {}
+}
+```
+
+</details>
+
+Try to fill in the "while loops" in the open and close leveraged position functions.
+
+Notice how the `openLeveragedPosition` receives a uint256 which represents the amount of ETH the caller wants left over as collateral after looping. If none is specified then the the loan will right at the liquidation threshold. The smallest movement in CORN going higher could cause you to be liquidated.
+
+The while loop should check add collateral to the `Lending` contract and then borrow the max amount of CORN. Then it should use the DEX to swap that CORN for more ETH. Then the loop should be good to go again. Just make sure you add a condition to check if the amount of ETH is equal to the reserve amount and if so, break out of the loop.
```

Comment:
"The while loop should ~~check~~ add collateral..."

Reply:
"...a condition to check if the amount of ETH is **less than or** equal to the reserve amount..."